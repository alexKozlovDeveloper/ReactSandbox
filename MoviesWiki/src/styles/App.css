h1 {
    color: #27aedb;
    text-align: center;
}

.body {
    background-color: dimgray;
}

.font {
    font-family: 'Graduate', cursive;
}