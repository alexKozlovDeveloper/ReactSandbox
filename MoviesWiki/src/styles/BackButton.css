.button {
    padding: 8px;
    font-size: 40px;
    background: url(https://static.thenounproject.com/png/95-200.png);
    min-width: 20px;
    min-height: 20px;
    background-size: contain;
    background-repeat: no-repeat;
    margin: 5px;
    cursor: pointer;
}