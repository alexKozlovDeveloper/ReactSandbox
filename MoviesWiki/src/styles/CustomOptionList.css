.container {
    height: 36px;
    padding: 5px;
}

.item {
    float: left;
    padding: 5px;
    margin: 2px;
    border: 1px solid #ccc;
    border-radius: 4px;
    font-family: cursive;
    cursor: pointer;
    background-color: #222327;
    color: white;
}

.isactive {
    background-color: gold;
}

.title {
    float: left;
    padding: 5px;
    margin: 2px;
    font-family: cursive;
    color: white;
}