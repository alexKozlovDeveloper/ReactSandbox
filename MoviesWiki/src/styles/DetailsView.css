.body {
    background-color: tomato;
    display: flex;
    padding-left: 30px;
    padding-right: 30px;
    padding-top: 10px;
    padding-bottom: 10px;
}

.imagecontainer {
    min-width: 150px;
}

.detailscontainer {
    padding: 10px;
}

.maintitlecontainer {
    display: flex;
}

.propertiescontainer {
    display: flex;
}

.descriptioncontainer {
    
}