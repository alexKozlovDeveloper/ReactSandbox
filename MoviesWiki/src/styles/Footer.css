.footer {
    bottom: 0;
    width: 100%;
    background: #333;
    padding: 10px 0;
    color: #fff;
    min-width: 500px;
}

.container {
    width: 80%;
    margin: 0 auto;
    text-align: center;
}