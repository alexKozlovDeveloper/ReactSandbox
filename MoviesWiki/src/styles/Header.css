.header {
    background-color: #7f8eba;
}

.headertitle {
    padding: 40px;
    font-family: cursive;
    color: aliceblue;
}