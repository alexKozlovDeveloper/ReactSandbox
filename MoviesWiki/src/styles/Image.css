.image {
    width: 100%;
}

.imagecontainer {

}