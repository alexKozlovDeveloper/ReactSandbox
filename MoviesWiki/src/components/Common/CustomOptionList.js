import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from '../../styles/CustomOptionList.css';

class CustomOptionList extends Component {
  constructor(props) {
    super(props);

    this.state = { selectedItem: `option${this.props.selectedIndex}` };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    this.setState({ selectedItem: e.currentTarget.attributes.name.nodeValue });
    this.props.updateFunc(e.currentTarget.attributes.value.nodeValue);
  }

  render() {
    const { options } = this.props.config;

    const optionsHtml = options.map((option, index) => {
      const name = `option${index}`;
      const itemStyles = styles.item + (this.state.selectedItem === name ? ` ${styles.isactive}` : '');

      return <div name={name} key={index} onClick={this.handleClick} className={itemStyles} value={option.value}>{option.title}</div>;
    });

    return (
      <div className={styles.container}>
        <div className={styles.title}>{this.props.config.title}</div>
        {optionsHtml}
      </div>
    );
  }
}

CustomOptionList.propTypes = {
  selectedIndex: PropTypes.string,
  updateFunc: PropTypes.func,
  config: PropTypes.shape({
    title: PropTypes.string,
    options: PropTypes.array,
  }),
};

export default CustomOptionList;
