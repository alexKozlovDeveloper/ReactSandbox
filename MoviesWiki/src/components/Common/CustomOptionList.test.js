import React from 'react';
import { mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import CustomOptionList from './CustomOptionList';
import styles from '../../styles/CustomOptionList.css';

configure({ adapter: new Adapter() });

describe('CustomOptionList component', () => {
  it('Rendering test', () => {
    const config = {
      title: 'Search by',
      selectedIndex: '0',
      options: [{
        title: 'Title',
        value: 'title',
      }, {
        title: 'Genre',
        value: 'genres',
      }],
    };

    const component = mount(<CustomOptionList config={config} updateFunc={() => { }} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('Click to second button', () => {
    const config = {
      title: 'Search by',
      selectedIndex: '0',
      options: [{
        title: 'Title',
        value: 'title',
      }, {
        title: 'Genre',
        value: 'genres',
      }],
    };

    const updateFunc = jest.fn();

    const component = mount(<CustomOptionList config={config} updateFunc={updateFunc} />);

    component.findWhere((node) => node.key() === '1').simulate('click');

    expect(component.findWhere((node) => node.key() === '1').hasClass(styles.isactive)).toBe(true);
    expect(component.state().selectedItem).toBe('option1');
    expect(updateFunc.mock.calls.length).toBe(1);
  });

  it('Click to first button', () => {
    const config = {
      title: 'Search by',
      selectedIndex: '1',
      options: [{
        title: 'Title',
        value: 'title',
      }, {
        title: 'Genre',
        value: 'genres',
      }],
    };

    const updateFunc = jest.fn();

    const component = mount(<CustomOptionList config={config} updateFunc={updateFunc} />);

    component.findWhere((node) => node.key() === '0').simulate('click');

    expect(component.findWhere((node) => node.key() === '0').hasClass(styles.isactive)).toBe(true);
    expect(component.state().selectedItem).toBe('option0');
    expect(updateFunc.mock.calls.length).toBe(1);
  });
});
