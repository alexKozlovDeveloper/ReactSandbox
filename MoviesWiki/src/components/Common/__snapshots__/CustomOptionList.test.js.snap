// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomOptionList component Rendering test 1`] = `
<CustomOptionList
  config={
    Object {
      "options": Array [
        Object {
          "title": "Title",
          "value": "title",
        },
        Object {
          "title": "Genre",
          "value": "genres",
        },
      ],
      "selectedIndex": "0",
      "title": "Search by",
    }
  }
  updateFunc={[Function]}
>
  <div
    className="container"
  >
    <div
      className="title"
    >
      Search by
    </div>
    <div
      className="item"
      key="0"
      name="option0"
      onClick={[Function]}
      value="title"
    >
      Title
    </div>
    <div
      className="item"
      key="1"
      name="option1"
      onClick={[Function]}
      value="genres"
    >
      Genre
    </div>
  </div>
</CustomOptionList>
`;
